nncf/common/pruning/operations.py

```diff
@@ -223,6 +225,66 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
         node.data['output_mask'] = result_mask
 
 
+class SplitPruningOp(BasePruningOp):
+    @classmethod
+    def match_multiple_output_masks(cls, output_masks: List[SymbolicMask], output_edges: List[NNCFGraphEdge],
+                                    chunk_axis: int) -> Dict['str', SymbolicMask]:
+        """
+        Match multiple input mask to each next nodes.
+
+        :param output_masks: Given output masks.
+        :param output_edges: Given output edges of the node.
+        :param chunk_axis: Given the axis on which operation was performed.
+        :return: Matched output mask for each next node.
+        """
+        result_masks = {}
+        tmp_output_masks = output_masks.copy()
+        tmp_output_masks_shape = [mask.shape[0] for mask in tmp_output_masks]
+        for edge in output_edges:
+            node_name = edge.to_node.node_name
+            idx = tmp_output_masks_shape.index(edge.tensor_shape[chunk_axis])
+            result_masks[node_name] = tmp_output_masks.pop(idx)
+            tmp_output_masks_shape.pop(idx)
+
+        return result_masks
+
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def generate_output_masks(cls, node: NNCFNode, graph: NNCFGraph,
+                              tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> Optional[NNCFTensor]:
+        """
+        Generate output mask from input masks for split/chunk operations.
+        If input mask is None return None
+
+        :param node: Node to determine its sources.
+        :param graph: NNCF graph to work with.
+        :param tensor_processor: Interface with tensor processing methods.
+        :return: Filled input masks.
+        """
+        input_masks = get_input_masks(node, graph)
+        if not input_masks:
+            return None
+        input_mask = input_masks[0]
+        chunk_axis = node.layer_attributes.axis
+
+        output_edges = graph.get_output_edges(node)
+        output_shapes = [edge.tensor_shape[chunk_axis] for edge in output_edges]
```

Comment:
Will it work as expected if axis =-1?
https://github.com/openvinotoolkit/nncf/pull/1244/files#diff-033daadd941b8d5ac07994ca370f200c90f8a8f5708edd49f0f20ed93d0756ebR88

Reply:
Still can't see how the issue with **identity split** is resolved. @sooahleex, could you please comment on that?

In my understanding, we should do nothing with mask in case of identity split, but instead we take the last shape, since axis = -1 here: https://github.com/openvinotoolkit/nncf/pull/1244/files#diff-033daadd941b8d5ac07994ca370f200c90f8a8f5708edd49f0f20ed93d0756ebR88
The logic looks not valid for that case, doesn't it?

Seems like you need additional logic if chunk_axis == -1

Reply:
I added the `axis=-1` case through `SplitModel` in tests/torch/pruning/helpers.py. Also, in this case, axis will not be set as `-1` but be `3`. It is the same logic of the axis setting in the concat method. If you think the current model is not enough, please let me know how to check it more.

Reply:
I think following logic is needed here:
```
if len(output_shapes) == 1: #if identity split detected
 return input_mask # propagate as is
...
```

Reply:
I've run all tests in `test_groups` and none of them set axis to -1. 
Added PR (https://github.com/sooahleex/nncf/pull/1) with the test case that handles identity split. 

Actually, it didn't pass before introducing Daniil's suggestion: https://github.com/openvinotoolkit/nncf/pull/1244#discussion_r956010338

Reply:
[Splitmodel](https://github.com/sooahleex/nncf/blob/548e8cfe76f6b7f78c2dcf3785d743b39ddb97fe/tests/torch/pruning/helpers.py#L616)
I added the model to cover `dim=-1` here. If it is not proper, please add to what you want.

Reply:
@sooahleex seems like there's a misunderstanding. 

I didn't mean `dim == -1` from this line: `y1, y2 = torch.chunk(x, chunks=2, dim=-1)`. 
I meant `axis == -1` from `get_split_axis` here: https://github.com/openvinotoolkit/nncf/pull/1244/files#diff-033daadd941b8d5ac07994ca370f200c90f8a8f5708edd49f0f20ed93d0756ebR88

Yes, both of them about chunk axis, but the meaning is different. 
-1 in the first line means last axis. 
-1 in the second will be only in the case of identity chunk when split do nothing and therefore there's no meaningful chunk axis. And logic with matching shapes was not valid if we take the last axis:
```python
output_shapes = [edge.tensor_shape[chunk_axis] for edge in output_edges]
if not input_mask.shape[0] == sum(output_shapes):
    return None
'''

It started to work with Daniil's suggestion only:
```
if len(output_shapes) == 1: #if identity split detected
 return input_mask # propagate as is
```

The model you mentioned doing ordinary chunk, not identity.
https://github.com/sooahleex/nncf/blob/548e8cfe76f6b7f78c2dcf3785d743b39ddb97fe/tests/torch/pruning/helpers.py#L616
That's why my PR makes sense to include.